core/string/src/display.rs

```diff
@@ -0,0 +1,25 @@
+//! Display implementations for [`crate::JsString`].
+use crate::{JsStr, JsStrVariant};
+use std::fmt;
+use std::fmt::Write;
+
+/// Display implementation for `JsString` that escapes unicode characters.
+#[derive(Debug)]
+pub struct JsStringDisplayEscaped<'a> {
+    pub(crate) inner: JsStr<'a>,
+}
+
+impl fmt::Display for JsStringDisplayEscaped<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.inner.variant() {
+            // SAFETY: `JsStrVariant::Latin1` is always valid latin1, so no need to check.
```

Comment:
Did you mean to say "latin1 is always valid utf8" here?

Reply:
Fixed. PTAL.